src/extensions/cellpack/state.ts

```diff
@@ -26,23 +30,165 @@ const ParseCellPack = PluginStateTransform.BuiltIn({
+                                fiber_counter_id++;
+                            } else {
+                                cell.mapping_ids[counter_id] = [comp_counter, iName];
+                                if (!surface.ingredients[iName].results) {surface.ingredients[iName].results = [];}
+                                counter_id++;
+                            }
+                        }
+                        comp_counter++;
+                    }
+                    if (interior) {
+                        packings.push({ name, location: 'interior', ingredients: interior.ingredients });
+                        for (iName in interior.ingredients){
+                            if (interior.ingredients[iName].ingtype === 'fiber') {
+                                cell.mapping_ids[-(fiber_counter_id + 1)] = [comp_counter, iName];
+                                if (!interior.ingredients[iName].nbCurve) interior.ingredients[iName].nbCurve = 0;
+                                fiber_counter_id++;
+                            } else {
+                                cell.mapping_ids[counter_id] = [comp_counter, iName];
+                                if (!interior.ingredients[iName].results) {interior.ingredients[iName].results = [];}
+                                counter_id++;
+                            }
+                        }
+                        comp_counter++;
+                    }
                 }
             }
-            if (cytoplasme) packings.push({ name: 'Cytoplasme', location: 'cytoplasme', ingredients: cytoplasme.ingredients });
-
+            if (params.modeFile && params.modeFile.file){
+                const model_data = await readFromFile(params.modeFile.file, 'binary').runInContext(ctx);// async ?
+                let numbers = new DataView(model_data.buffer);
+                let ninst = getFloatValue(numbers, 0);
+                let npoints = getFloatValue(numbers, 4);
+                let ncurve = getFloatValue(numbers, 8);
+                /*
+                console.log("Parse CellPack");
+                console.log("ninst ",ninst);
+                console.log("npoints ",npoints);
+                console.log("ncurve ",ncurve);
+                */
+                let pos = new Float32Array();
+                let quat = new Float32Array();
+                let ctr_pos = new Float32Array();
+                // let ctr_norm = new Float32Array();
+                let ctr_info = new Float32Array();
+                let curve_ids = new Float32Array();
+                // let ctrl_pts= new Float32Array();
+                // let ctrl_normal= new Float32Array();
+                // let ctrl_info= new Float32Array();//#curve_id, curve_type, angle, uLength
+                let offset = 12;
+                if (ninst !== 0){
+                    pos = new Float32Array(model_data.buffer, offset, ninst * 4);offset += ninst * 4 * 4;
```

Comment:
What endianess are your data? There are ``isLittleEndian()``/``IsNativeEndianLittle`` and ``flipByteOrder`` functions in Mol* that you can use to make sure the data are loaded correctly.

Reply:
The other way around (you want to convert into the native endian). Something like this (I hope :)):

```ts
if (!IsNativeEndianLittle) {
    // only flip the byte order on the current view
    const buffer = flipByteOrder(new Uint8Array(data.buffer, offset, 4 * size), 4);
    return new Float32Array(buffer);
} else {
    return new Float32Array(data.buffer, offset, size);
}
```

Reply:
great thanks, I made the change.